engine/dlib/src/dlib/job_thread.cpp

```diff
@@ -115,16 +114,22 @@ static void JobThread(void* _ctx)
     }
 }
 #else
-static void UpdateSingleThread(JobThreadContext* ctx)
+static void UpdateSingleThread(JobThreadContext* ctx, uint64_t max_time)
 {
-    DM_PROFILE("SingleThread");
-    if (ctx->m_Work.Empty())
-        return;
-    // TODO: Perhaps time scope a number of items!
-    JobItem item = ctx->m_Work.Pop();
+    while (!ctx->m_Work.Empty())
+    {
+        uint64_t tstart = dmTime::GetTime();
+        JobItem item = ctx->m_Work.Pop();
+
+        item.m_Result = item.m_Process(item.m_Context, item.m_Data);
+        PutDone(ctx, &item);
 
-    item.m_Result = item.m_Process(item.m_Context, item.m_Data);
-    PutDone(ctx, &item);
+        uint64_t tend = dmTime::GetTime();
+        if ((tend-tstart) > max_time)
```

Comment:
If max_time == 0, then it still behaves as before: Process one job per update.

Reply:
I think it's worth to accumulate all time spent on job process and compare accumulated time to `max_time`. Because it can be lots of small jobs which doesn't exceed the `max_time` but in sum it takes much more time than `max_time`. For example,
* 10 jobs. Each takes 0.05s to process
* max_time == 0.1s
* total time is 0.05 * 10 = 0.5s which is much more than 0.1s

Reply:
Hmm, yeah that was the intent, but I wonder how I got it turned around in my head, as that's not how it's implemented?!

Thanks for the heads up!

Reply:
Yeah, I didn't notice it because the workload I tested with was way faster than the limit I set :)